nmstatectl/nmstatectl.py

```diff
@@ -50,12 +54,20 @@ def setup_subcommand_set(subparsers):
 
 
 def show(args):
-    print(json.dumps(netinfo.show(), indent=4, sort_keys=True,
-                     separators=(',', ': ')))
+    state = netinfo.show()
+    if args.yaml:
+        print(yaml.dump(state, default_flow_style=False))
+    else:
+        print(json.dumps(state, indent=4, sort_keys=True,
```

Comment:
It will also be nice to have a --json flag, to output a "pure" json format (for callers to consume).

Reply:
I guess you mean no pretty printing with "pure". Callers can consume the pretty-printed format AFAIK, however I changed it to do this only on ttys. `nmstateclt show | cat` will show you the raw format. Is this what you meant? I did not squash it, yet, so you can review the changes.

Reply:
Other networking cli tools (ovs-vsctl for example) print a raw json output (non pretty as you said).
Not sure if it matters to a consumer, and I would expect it to use the lib and not the ctl anyway.
Your solution is nice, but in the code it looks ugly (too many if/s, too much options ...) lets take it out and assume the callers can consume it as you mentioned.